### PR TITLE
Feature/166057647 circleci precommit

### PR DESCRIPTION
**Pull request by ahobson**
[#166057647](https://www.pivotaltracker.com/story/show/166057647)

Use markdownlint as a pre-commit hook and run that pre-commit in circleci to ensure the docs remain sparkly clean.

Convert the docs to markdownlint approved syntax.

Prefer `*` to `-` when a conflict is found

### CODE REVIEW

**Comment by sojeri** [comment]
can you update the PR description to include a link to the pivotal story?

File: developing/cycle/tracker.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # [Development Cycle](./README.md) / Pivotal Tracker
 
 ## Overview
+
 We use Pivotal Tracker to track tasks and estimate how much time each task will take.
 
-* 🔒[How to organize a Pivotal project](https://docs.google.com/document/d/1abmxQdYol6Zi3kR3U__JNJl3uPdKF9WPYWfu75AkZ50/edit#heading=h.yyb7wuihugn8)
-* [Pivotal workflow](https://www.pivotaltracker.com/help/articles/workflow_overview/)
+- [How to organize a Pivotal project](https://docs.google.com/document/d/1abmxQdYol6Zi3kR3U__JNJl3uPdKF9WPYWfu75AkZ50/edit#heading=h.yyb7wuihugn8)
```
Comment (sojeri) [inline developing/cycle/tracker.md]: markdownlint has deleted the lock emoji indicating the pivotal board is not public

**Comment by ntwyman** [comment]
Can we get an overall comment or introduction to give context for this? I am guessing that it is adding CircleCI to check the repo but that's a guess.

File: developing/cycle/tracker.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # [Development Cycle](./README.md) / Pivotal Tracker
 
 ## Overview
+
 We use Pivotal Tracker to track tasks and estimate how much time each task will take.
 
-* 🔒[How to organize a Pivotal project](https://docs.google.com/document/d/1abmxQdYol6Zi3kR3U__JNJl3uPdKF9WPYWfu75AkZ50/edit#heading=h.yyb7wuihugn8)
-* [Pivotal workflow](https://www.pivotaltracker.com/help/articles/workflow_overview/)
+- 🔒[How to organize a Pivotal project](https://docs.google.com/document/d/1abmxQdYol6Zi3kR3U__JNJl3uPdKF9WPYWfu75AkZ50/edit#heading=h.yyb7wuihugn8)
```
Comment (ntwyman) [inline developing/cycle/tracker.md]: This PR seems to be consistently changing * to -. Why? Hard to know if this is intentional without context

Comment (ahobson) [reply developing/cycle/tracker.md]: My apologies for not doing a better job of explaining context.  markdownlint complained about using `*` for bullets instead of `-` so I changed it.

Comment (brainsik) [reply developing/cycle/tracker.md]: That's … odd. We use `*` on pretty much all our other docs. I'm confused as to why it was complaining about these …

Comment (ahobson) [reply developing/cycle/tracker.md]: I ... I can't explain what happened, so some kind of user error must have occurred, I just don't know what.  Let me Try Again with a much smaller set of changes.

Thank you for paying attention to the details.

Comment (ahobson) [reply developing/cycle/tracker.md]: Ok, here's an example: ```
developing/languages/README.md: 35: MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
```

What I misunderstood about the error is that it just means the list character needs to be consistent in the document, not that it needs to be changed everywhere.  Apologies.  Updated PR shortly.

Comment (ahobson) [reply developing/cycle/tracker.md]: I've redone the PR to be less invasive and updated the PR description to hopefully better describe what is going on.

Apologies for the sloppiness.
